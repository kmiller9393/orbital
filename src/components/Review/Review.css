.review-button-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 1.5rem; }

.review-header-container {
  padding-right: 6.5rem; }

.review-button {
  background-color: #009c92;
  border-color: #009c92;
  border-radius: 5px;
  height: 45px;
  margin-bottom: 1rem;
  width: 175px; }

.back-arrow {
  height: 40px;
  padding: 0rem 4rem 0.5rem 0rem; }

.review-header {
  color: #f7f7f7;
  margin-top: 2.5rem; }

.review-header-image {
  height: 55px;
  margin-top: 1.5rem;
  width: 125px; }

.review-page-link {
  color: #f7f7f7;
  font-size: 1.3rem;
  text-decoration: none; }

.review-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  font-family: 'Chalkboard SE', 'Times New Roman', serif; }
