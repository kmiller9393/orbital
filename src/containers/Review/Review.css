.loading-gif {
  max-height: 50px;
  max-width: 50px; }

.review-button-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 2.5rem; }

.review-header-container {
  padding-right: 6.5rem; }

.review-button {
  background-color: #47a234;
  border-color: #47a234;
  border-radius: 5px;
  box-shadow: 4px 8px 15px rgba(0, 0, 0, 0.3);
  cursor: pointer;
  height: 45px;
  margin-bottom: 1rem;
  text-shadow: 2px 4px 3px rgba(0, 0, 0, 0.3);
  width: 175px; }

.review-button:active {
  background-color: #4fb83b; }

.back-arrow {
  height: 40px;
  padding: 0rem 4rem 0.5rem 0rem; }

.review-header {
  color: #f7f7f7;
  margin-top: 2.5rem;
  text-shadow: 2px 4px 3px rgba(0, 0, 0, 0.7); }

.review-header-image {
  height: 55px;
  margin-top: 1.5rem;
  width: 125px; }

.review-page-link {
  color: #f7f7f7;
  font-size: 1.3rem;
  text-decoration: none; }

.review-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  height: 100%; }

.center {
  justify-content: center; }
