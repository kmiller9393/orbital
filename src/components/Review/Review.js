import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import orbital from '../../images/orbital-header.svg';
import './Review.css';

export default class Review extends Component {
  render() {
    return (
      <div className="review-container">
        <NavLink to="/">
          <img className="review-header-image" src={orbital} alt="Orbital" />
        </NavLink>
        <h2 className="review-header">Weekly Reviews</h2>
        <div className="review-button-container">
          <button className="review-button">
            <NavLink className="review-page-link" to="/review-one">
              Week 1
            </NavLink>
          </button>
          <button className="review-button">
            <NavLink className="review-page-link" to="/review-two">
              Week 2
            </NavLink>
          </button>
          <button className="review-button">
            <NavLink className="review-page-link" to="/review-three">
              Week 3
            </NavLink>
          </button>
          <button className="review-button">
            <NavLink className="review-page-link" to="/review-four">
              Week 4
            </NavLink>
          </button>
          <button className="review-button">
            <NavLink className="review-page-link" to="/review-five">
              Week 5
            </NavLink>
          </button>
        </div>
      </div>
    );
  }
}
