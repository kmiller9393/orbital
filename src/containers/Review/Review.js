import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import orbital from '../../images/orbital-header.svg';
import backArrow from '../../images/left-arrow.svg';
import loadingGif from '../../images/loading-globe.gif';
import './Review.css';

class Review extends Component {
  render() {
    const { glossary } = this.props;

    return (
      <div className="review-container">
        {!glossary.length ? (
          <img className="loading-gif" src={loadingGif} alt="loading" />
        ) : (
          ''
        )}
        {glossary.length ? (
          <div className="review-header-container">
            <NavLink to="/">
              <img className="back-arrow" src={backArrow} alt="Navigate Back" />
            </NavLink>
            <NavLink to="/">
              <img
                className="review-header-image"
                src={orbital}
                alt="Orbital"
              />
            </NavLink>
          </div>
        ) : (
          ''
        )}
        {glossary.length ? (
          <h2 className="review-header">Weekly Reviews</h2>
        ) : (
          ''
        )}
        {glossary.length ? (
          <div className="review-button-container">
            <button className="review-button">
              <NavLink className="review-page-link" to="/review-one">
                Week 1
              </NavLink>
            </button>
            <button className="review-button">
              <NavLink className="review-page-link" to="/review-two">
                Week 2
              </NavLink>
            </button>
            <button className="review-button">
              <NavLink className="review-page-link" to="/review-three">
                Week 3
              </NavLink>
            </button>
            <button className="review-button">
              <NavLink className="review-page-link" to="/review-four">
                Week 4
              </NavLink>
            </button>
            <button className="review-button">
              <NavLink className="review-page-link" to="/review-five">
                Week 5
              </NavLink>
            </button>
          </div>
        ) : (
          ''
        )}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  glossary: state.glossary
});

export default connect(
  mapStateToProps,
  null
)(Review);
