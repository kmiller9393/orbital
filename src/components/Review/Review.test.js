import React from 'react';
import { shallow } from 'enzyme';
import Review from './Review';

describe('Review', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Review />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
