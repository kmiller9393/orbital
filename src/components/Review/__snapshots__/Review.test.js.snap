// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Review />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
>
          <img
                    alt="Orbital"
                    className="review-header-image"
                    src="orbital-header.svg"
          />
</NavLink>,
        <h2
          className="review-header"
>
          Weekly Reviews
</h2>,
        <div
          className="review-button-container"
>
          <button
                    className="review-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="review-page-link"
                              to="/review-one"
                    >
                              Week 1
                    </NavLink>
          </button>
          <button
                    className="review-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="review-page-link"
                              to="/review-two"
                    >
                              Week 2
                    </NavLink>
          </button>
          <button
                    className="review-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="review-page-link"
                              to="/review-three"
                    >
                              Week 3
                    </NavLink>
          </button>
          <button
                    className="review-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="review-page-link"
                              to="/review-four"
                    >
                              Week 4
                    </NavLink>
          </button>
          <button
                    className="review-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="review-page-link"
                              to="/review-five"
                    >
                              Week 5
                    </NavLink>
          </button>
</div>,
      ],
      "className": "review-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <img
            alt="Orbital"
            className="review-header-image"
            src="orbital-header.svg"
/>,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Orbital",
            "className": "review-header-image",
            "src": "orbital-header.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Weekly Reviews",
          "className": "review-header",
        },
        "ref": null,
        "rendered": "Weekly Reviews",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="review-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="review-page-link"
                            to="/review-one"
              >
                            Week 1
              </NavLink>
</button>,
            <button
              className="review-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="review-page-link"
                            to="/review-two"
              >
                            Week 2
              </NavLink>
</button>,
            <button
              className="review-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="review-page-link"
                            to="/review-three"
              >
                            Week 3
              </NavLink>
</button>,
            <button
              className="review-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="review-page-link"
                            to="/review-four"
              >
                            Week 4
              </NavLink>
</button>,
            <button
              className="review-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="review-page-link"
                            to="/review-five"
              >
                            Week 5
              </NavLink>
</button>,
          ],
          "className": "review-button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="review-page-link"
                to="/review-one"
>
                Week 1
</NavLink>,
              "className": "review-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 1",
                "className": "review-page-link",
                "to": "/review-one",
              },
              "ref": null,
              "rendered": "Week 1",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="review-page-link"
                to="/review-two"
>
                Week 2
</NavLink>,
              "className": "review-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 2",
                "className": "review-page-link",
                "to": "/review-two",
              },
              "ref": null,
              "rendered": "Week 2",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="review-page-link"
                to="/review-three"
>
                Week 3
</NavLink>,
              "className": "review-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 3",
                "className": "review-page-link",
                "to": "/review-three",
              },
              "ref": null,
              "rendered": "Week 3",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="review-page-link"
                to="/review-four"
>
                Week 4
</NavLink>,
              "className": "review-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 4",
                "className": "review-page-link",
                "to": "/review-four",
              },
              "ref": null,
              "rendered": "Week 4",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="review-page-link"
                to="/review-five"
>
                Week 5
</NavLink>,
              "className": "review-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Week 5",
                "className": "review-page-link",
                "to": "/review-five",
              },
              "ref": null,
              "rendered": "Week 5",
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
>
            <img
                        alt="Orbital"
                        className="review-header-image"
                        src="orbital-header.svg"
            />
</NavLink>,
          <h2
            className="review-header"
>
            Weekly Reviews
</h2>,
          <div
            className="review-button-container"
>
            <button
                        className="review-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="review-page-link"
                                    to="/review-one"
                        >
                                    Week 1
                        </NavLink>
            </button>
            <button
                        className="review-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="review-page-link"
                                    to="/review-two"
                        >
                                    Week 2
                        </NavLink>
            </button>
            <button
                        className="review-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="review-page-link"
                                    to="/review-three"
                        >
                                    Week 3
                        </NavLink>
            </button>
            <button
                        className="review-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="review-page-link"
                                    to="/review-four"
                        >
                                    Week 4
                        </NavLink>
            </button>
            <button
                        className="review-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="review-page-link"
                                    to="/review-five"
                        >
                                    Week 5
                        </NavLink>
            </button>
</div>,
        ],
        "className": "review-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <img
              alt="Orbital"
              className="review-header-image"
              src="orbital-header.svg"
/>,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Orbital",
              "className": "review-header-image",
              "src": "orbital-header.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Weekly Reviews",
            "className": "review-header",
          },
          "ref": null,
          "rendered": "Weekly Reviews",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="review-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="review-page-link"
                                to="/review-one"
                >
                                Week 1
                </NavLink>
</button>,
              <button
                className="review-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="review-page-link"
                                to="/review-two"
                >
                                Week 2
                </NavLink>
</button>,
              <button
                className="review-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="review-page-link"
                                to="/review-three"
                >
                                Week 3
                </NavLink>
</button>,
              <button
                className="review-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="review-page-link"
                                to="/review-four"
                >
                                Week 4
                </NavLink>
</button>,
              <button
                className="review-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="review-page-link"
                                to="/review-five"
                >
                                Week 5
                </NavLink>
</button>,
            ],
            "className": "review-button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="review-page-link"
                  to="/review-one"
>
                  Week 1
</NavLink>,
                "className": "review-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 1",
                  "className": "review-page-link",
                  "to": "/review-one",
                },
                "ref": null,
                "rendered": "Week 1",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="review-page-link"
                  to="/review-two"
>
                  Week 2
</NavLink>,
                "className": "review-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 2",
                  "className": "review-page-link",
                  "to": "/review-two",
                },
                "ref": null,
                "rendered": "Week 2",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="review-page-link"
                  to="/review-three"
>
                  Week 3
</NavLink>,
                "className": "review-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 3",
                  "className": "review-page-link",
                  "to": "/review-three",
                },
                "ref": null,
                "rendered": "Week 3",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="review-page-link"
                  to="/review-four"
>
                  Week 4
</NavLink>,
                "className": "review-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 4",
                  "className": "review-page-link",
                  "to": "/review-four",
                },
                "ref": null,
                "rendered": "Week 4",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="review-page-link"
                  to="/review-five"
>
                  Week 5
</NavLink>,
                "className": "review-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Week 5",
                  "className": "review-page-link",
                  "to": "/review-five",
                },
                "ref": null,
                "rendered": "Week 5",
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
